src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,198 @@
+package frc.robot.subsystems.conveyor;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorSensorV3;
+import edu.wpi.first.wpilibj.*;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.LinkedList;
+
+import static frc.robot.Ports.Conveyor.INVERSION;
+
+public class Conveyor extends SubsystemBase {
+    private static Conveyor INSTANCE = null;
+    private final WPI_TalonFX motor = new WPI_TalonFX(Ports.Conveyor.MOTOR);
+    private final Deque<String> position = new LinkedList<>();
+    private final ColorSensorV3 colorSensorIntake = new ColorSensorV3(I2C.Port.kOnboard);
+    private final DigitalInput postFlapBeam = new DigitalInput(Ports.Conveyor.POST_FLAP_BEAM_BREAKER);
+    private final DigitalInput preFlapBeam = new DigitalInput(Ports.Conveyor.PRE_FLAP_BEAM_BREAKER);
+    private final Solenoid flap = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Conveyor.SOLENOID);
+    private final ColorMatch match = new ColorMatch();
+    private DriverStation.Alliance lastSeenColor = DriverStation.Alliance.Invalid;
+    private boolean lastPostFlapBeamInput = true;
+
+    private Conveyor() {
+        motor.setInverted(INVERSION);
+        match.addColorMatch(Constants.Conveyor.RED);
+        match.addColorMatch(Constants.Conveyor.BLUE);
+        match.addColorMatch(Constants.Conveyor.NONE);

```

Comment:
Another, more concise option:
- in the constants file:
```java
 public static final Color[] COLORS =
        {new Color(0, 0, 0),
                new Color(0, 0, 0),
                new Color(0, 0, 0)};
```
- Here:
```java
for (Color color : Constants.Conveyor.COLORS) {
    match.addColorMatch(color);
}
```
That way, if you need to add colors, you only need to modify the `COLORS` array.

---
P.S.: I just noticed you used the colors individually, so this suggestion might not be as helpful as it could. However, it's a nice design pattern you should keep in mind in the future because it can reduce code repetition.